commit da970d24ecfcf67f89a9532f3a53ade8cb1131ed
Author: Bert Vermeulen <bert@biot.com>
Date:   Fri Aug 30 13:57:15 2013 +0200

    rigol-ds1xx2: newer Linux kernels have USBTMC in /sys/class/usbmisc

diff --git a/hardware/rigol-ds1xx2/api.c b/hardware/rigol-ds1xx2/api.c
index e6e923c..000de31 100644
--- a/hardware/rigol-ds1xx2/api.c
+++ b/hardware/rigol-ds1xx2/api.c
@@ -307,8 +307,9 @@ static GSList *scan(GSList *options)
 		if (probe_port(port, &devices) == SR_ERR_MALLOC)
 			return NULL;
 	} else {
-		if (!(dir = g_dir_open("/sys/class/usb/", 0, NULL)))
-			return NULL;
+		if (!(dir = g_dir_open("/sys/class/usbmisc/", 0, NULL)))
+			if (!(dir = g_dir_open("/sys/class/usb/", 0, NULL)))
+				return NULL;
 		while ((dev_name = g_dir_read_name(dir))) {
 			if (strncmp(dev_name, "usbtmc", 6))
 				continue;
